### PR TITLE
[CF-612] Creating cinder volume types

### PR DESCRIPTION
**Pull request by SVilgelm**
Extend generate load to add cinder volume types.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Some checks failed.
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry%20checks/136/

**Comment by MirantisWorkloadMobility** [comment]
Some checks failed.
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry%20checks/137/

**Comment by MirantisWorkloadMobility** [comment]
Some checks failed.
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry%20checks/145/

**Comment by MirantisWorkloadMobility** [comment]
Some checks failed.
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry%20checks/153/

**Comment by MirantisWorkloadMobility** [comment]
Some checks failed.
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry%20checks/154/

**Comment by MirantisWorkloadMobility** [comment]
Some checks failed.
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry%20checks/156/

**Comment by MirantisWorkloadMobility** [comment]
Some checks failed.
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry%20checks/158/

**Comment by MirantisWorkloadMobility** [comment]
Some checks failed.
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry%20checks/163/

**Comment by MirantisWorkloadMobility** [comment]
Some checks failed.
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry%20checks/164/

**Comment by MirantisWorkloadMobility** [comment]
Some checks failed.
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry%20checks/167/

**Comment by MirantisWorkloadMobility** [comment]
Some checks failed.
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry%20checks/170/

File: cloudferry/lib/os/storage/cinder_storage.py [hunk 1]
```diff
@@ -316,6 +316,9 @@ def create_volume_from_volume(self, volume, tenant_id):
 
         az = self.attr_override.get_attr(volume, 'availability_zone')
         volume_type = self.attr_override.get_attr(volume, 'volume_type')
+        if volume_type != volume['volume_type']:
+            LOG.debug("Volume type '%s' of volume '%s' is overridden for '%s'",
+                      volume['volume_type'], volume['id'], volume_type)
```
Comment (roman-verchikov) [inline cloudferry/lib/os/storage/cinder_storage.py]: Let's move this log to `attr_override.get_attr()` method, or remove it completely, it doesn't cover a lot of cases.

File: cloudferry_devlab/cloudferry_devlab/generate_load.py [hunk 2]
```diff
@@ -691,6 +692,23 @@ def create_security_group_rule(self, group_id, tenant_id, protocol='tcp',
...
+    def create_cinder_volume_types(self, volume_types_list):
+        self.log.debug("Creating cinder volume types on '%s' cloud",
+                       self.cloud_prefix)
+        for volume_type in volume_types_list:
+            try:
+                vt = self.cinderclient.volume_types.create(volume_type['name'])
+                vt.set_keys(volume_type['metadata'])
+            except cd_exceptions.ClientException as e:
+                self.log.debug(e)
+                self.log.warning("Volume type '%s' already exists",
```
Comment (roman-verchikov) [inline cloudferry_devlab/cloudferry_devlab/generate_load.py]: `ClientException` probably represents more than one exception type, let's modify warning message to `"Failed to create volume type '%s', error: %s", error`

File: cloudferry/lib/os/storage/cinder_storage.py (hunk 1 above)
Comment (SVilgelm) [inline cloudferry/lib/os/storage/cinder_storage.py]: done

File: cloudferry_devlab/cloudferry_devlab/generate_load.py (hunk 2 above)
Comment (SVilgelm) [inline cloudferry_devlab/cloudferry_devlab/generate_load.py]: done

**Comment by MirantisWorkloadMobility** [comment]
Some checks failed.
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry%20checks/179/

**Pull request merged**